### PR TITLE
Porting UPP for gfs v16  on wcoss2

### LINKED ISSUE #363: Port UPP for GFS V16 to WCOSS2
Update the branch release/gfs_v16 to support GFS V16 on WCOSS2.

Related issue:

- https://github.com/NOAA-EMC/global-workflow/issues/398

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
The changes are added for porting UPP for gfs v16 on wcoss2.

### CODE REVIEW

**Comment by aerorahul** [comment]
@WenMeng-NOAA why are manage_externals changes required here?

**Comment by WenMeng-NOAA** [comment]
@aerorahul @KateFriedman-NOAA found a new version of manage_externals is needed for the global-workflow checking gtg code.

**Comment by KateFriedman-NOAA** [comment]
> @aerorahul @KateFriedman-NOAA found a new version of manage_externals is needed for the global-workflow checking gtg code.

@WenMeng-NOAA Can the GTG checkout happen without manage_externals? I use whatever method you provide on the workflow side to do the checkouts. We're trying to move away from manage_externals now for a few reasons. Can we clone the GTG with submodule commands instead? Thanks!

File: jobs/J_NCEPPOST [hunk 1]
```diff
@@ -27,7 +27,7 @@ export KEEPDATA=${KEEPDATA:-NO}
 ####################################
 # File To Log Msgs
 ####################################
-export jlogfile=${jlogfile:-${COMROOT}/logs/jlogfiles/jlogfile.${job}.${pid}}
+#export jlogfile=${jlogfile:-${COMROOT}/logs/jlogfiles/jlogfile.${job}.${pid}}
```
Comment (KateFriedman-NOAA) [inline jobs/J_NCEPPOST]: Perhaps just remove this line now that we're retiring the use of jlogfiles?

**Review by KateFriedman-NOAA — CHANGES_REQUESTED** [review]
Non-manage-externals-related changes look good and cover the changes Wen and I discussed when I initially tested the post jobs on WCOSS2. I am rebuilding my copy of the UPP with the recent LUA updates and will do the following for this PR:

1. retest post jobs on Dogwood with updated UPP copy and have Wen look at output
2. update my copy of the UPP to use submodules instead of manage_externals; will provide needed updates for Wen to incorporate

Stay tuned...

**Comment by WenMeng-NOAA** [comment]
> > @aerorahul @KateFriedman-NOAA found a new version of manage_externals is needed for the global-workflow checking gtg code.
> 
> @WenMeng-NOAA Can the GTG checkout happen without manage_externals? I use whatever method you provide on the workflow side to do the checkouts. We're trying to move away from manage_externals now for a few reasons. Can we clone the GTG with submodule commands instead? Thanks!

@KateFriedman-NOAA Yes, @aerorahul has been helping me on configurating gtg code checking out as a submodule.

**Comment by KateFriedman-NOAA** [comment]
@WenMeng-NOAA Please see the following log from rebuilding the UPP after recent updates, please confirm the build went as intended:

/lfs/h2/emc/global/noscrub/Kate.Friedman/git/feature-ops-wcoss2/sorc/logs/build_ncep_post_test5.log

Thanks!

Note: this is before I make any changes for the GTG checkout

File: jobs/J_NCEPPOST (hunk 1 above)
Comment (WenMeng-NOAA) [reply jobs/J_NCEPPOST]: @KateFriedman-NOAA done!

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA Please see the following log from rebuilding the UPP after recent updates, please confirm the build went as intended:
> 
> /lfs/h2/emc/global/noscrub/Kate.Friedman/git/feature-ops-wcoss2/sorc/logs/build_ncep_post_test5.log
> 
> Thanks!
> 
> Note: this is before I make any changes for the GTG checkout

@KateFriedman-NOAA Your build log look good to me. Thanks!

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA Your build log look good to me. Thanks!

Thanks for confirming @WenMeng-NOAA ! I'm working on the patch to retire manage_externals and use submodules instead for the GTG checkout. Will provide the patch to include in this PR when ready.

**Comment by KateFriedman-NOAA** [comment]
@WenMeng-NOAA A quick update... @aerorahul and I are still working on the patch (ironing out a detail) but I have removed manage_externals in my fork copy of the `post_gfsv16_wcoss2` branch and added the initial submodule setup. See my changes thus far at this link:

https://github.com/WenMeng-NOAA/UPP/compare/post_gfsv16_wcoss2...KateFriedman-NOAA:post_gfsv16_wcoss2

I hacked things on the workflow side a tad to get the right tag version of the GTG checked out (the detail to iron out) and tested the build with GTG changes in place. Please see the following log and confirm that my build with GTG worked:

/lfs/h2/emc/global/noscrub/Kate.Friedman/git/feature-ops-wcoss2/sorc/logs/build_ncep_post_test6_gtg.log

I will submit a PR into your fork `post_gfsv16_wcoss2` branch with the patch when ready. Thanks!

**Comment by WenMeng-NOAA** [comment]
> /lfs/h2/emc/global/noscrub/Kate.Friedman/git/feature-ops-wcoss2/sorc/logs/build_ncep_post_test6_gtg.log

Your build log looks good to me. Thanks!

**Comment by KateFriedman-NOAA** [comment]
> Your build log looks good to me. Thanks!

Thanks @WenMeng-NOAA ! I just reran all of my gdaspost* jobs and the gfspost023 job in my Dogwood test. Please see the following logs and output and let me know if it looks ok:

gdaspost*:
/lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/logs/2021082400/gdaspost0*log
/lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/gdas.20210824/00/atmos

gfspost023 (just redid the f213 files):
/lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/logs/2021082400/gfspost023.log
/lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/gfs.20210824/00/atmos/*f213*

FYI, I will be running a cleaner fresh test next week for everyone to examine, so I'll send logs/output to look at again when I do that test. Thanks!

**Comment by WenMeng-NOAA** [comment]
> > Your build log looks good to me. Thanks!
> 
> Thanks @WenMeng-NOAA ! I just reran all of my gdaspost* jobs and the gfspost023 job in my Dogwood test. Please see the following logs and output and let me know if it looks ok:
> 
> gdaspost*: /lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/logs/2021082400/gdaspost0*log /lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/gdas.20210824/00/atmos
> 
> gfspost023 (just redid the f213 files): /lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/logs/2021082400/gfspost023.log /lfs/h2/emc/ptmp/Kate.Friedman/comrot/testcyc768b/gfs.20210824/00/atmos/_f213_
> 
> FYI, I will be running a cleaner fresh test next week for everyone to examine, so I'll send logs/output to look at again when I do that test. Thanks!

@KateFriedman-NOAA The runtime logs and UPP datasets look good to me. Thanks!

**Comment by KateFriedman-NOAA** [comment]
@WenMeng-NOAA I submitted [PR #5](https://github.com/WenMeng-NOAA/UPP/pull/5) from my fork to your fork with updates for converting the GTG checkout from manage_externals to git submodules. I wasn't able to add reviewers or assignees or labels.

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA Your PR looks good to me. Would you like to have someone else review it?

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA Your PR looks good to me. Would you like to have someone else review it?

Nope, since it works for you and both @aerorahul and myself tested it, it should be good to submit my PR into your branch. Thanks!

**Comment by WenMeng-NOAA** [comment]
@KateFriedman-NOAA and @aerorahul It looks the branch is working for inline and offline post in Kate's GFS V16 end-to-end testing on WCOSS2. Do you have more comments  on it? Thanks!

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks good and post jobs in GFS tests are working as intended. Thanks!

**Review by aerorahul — APPROVED** [review]
Defer to @KateFriedman-NOAA. No further comments from me.

**Comment by KateFriedman-NOAA** [comment]
@WenMeng-NOAA If you could look at the post output in my free-forecast test that ran overnight and then, assuming it looks good, I'm good for this to be submitted. See this g-w issue comment with output details: https://github.com/NOAA-EMC/global-workflow/issues/399#issuecomment-989998164 . Thanks!

**Pull request merged**